cfme/common/host_views.py

```diff
@@ -276,6 +276,5 @@ class ProviderAllHostsView(HostsView):
     def is_displayed(self):
         return (
             self.navigation.currently_selected == ["Compute", "Infrastructure", "Providers"] and
-            self.title.text == "{} (All Managed Hosts)".format(
-                self.context["object"].filters['parent'].name)
+            self.title.text == "{} (All Managed Hosts)".format(self.context["object"].name)
```

Comment:
I don't think this fix will work - but you're absolutely on the right path.

So this view is used in two cases (at least) - `HostsCollection` 'All' nav destination in `cfme/infrastructure/host.py` and for `InfraProvider` 'Hosts' destination in `cfme/infrastructure/provider/__init__.py`

In the HostsCollection destination, the view is only selected when a parent or provider filter is applied to the collection.

Here, we can't assume that the contextual object is the provider - just like we can't assume its the HostsCollection either.  The current code assumes there's a parent filter, your change assumes that the context object is a provider.

This should check for the presence of a parent/provider filter, just like the VIEW property in HostsCollection 'All' nav destination.  The other thing to pay attention to is how the view may be used in a test.

Reply:
@mshriver Thank you for the explanation. I see it's a bit more complicated than I originally thought.